hpx/util/cache_aligned_data.hpp

```diff
@@ -55,6 +55,84 @@ namespace hpx { namespace util
             (sizeof(Data) % threads::get_cache_line_size())];
     };
 #endif
+
+    ///////////////////////////////////////////////////////////////////////////
+    // special struct to ensure cache line alignment of data
+    // consisting of multiple elements (a tuple in effect)
+    // The tuple is cache aligned and fully occupies 1 or more cache lines
+    // use get<0>, get<1>, ... get<N> to access the internal elements
+#if defined(HPX_HAVE_CXX11_ALIGNAS) && defined(HPX_HAVE_CXX17_ALIGNED_NEW) &&  \
+    defined(HPX_HAVE_CXX17_FOLD_EXPRESSIONS) && !defined(__NVCC__)
+    template <typename ... Data>
+    struct alignas(threads::get_cache_line_size()) cache_line_multidata
+    {
+        template <typename ... T>
+        cache_line_multidata(T...ts) : data_(ts...) {};
+
+        template <std::size_t I>
+        auto &get() {
+            return std::get<I>(data_);
+        }
+
+        template <std::size_t I>
+        auto const &get() const {
+            return std::get<I>(data_);
+        }
+
+        // pad to cache line size bytes
+        std::tuple<Data...> data_;
+        char cacheline_pad[(0 + ... + sizeof(Data)) % threads::get_cache_line_size()];
```

Comment:
I think `sizeof...(Data)` should work here. Then you should be able to get rid of the other implementation.

Reply:
`sizeof()...` on its own will give you the pack size, but not the bytes used by the elements, that's the reason for the long-winded template stuff. the fold operator does it very nicely for c++17

NB. force pushed because I had
`(0 + ... + sizeof(Data)) % threads::get_cache_line_size()`
instead of 
`threads::get_cache_line_size() - ((0 + ... + sizeof(Data)) % threads::get_cache_line_size())`
in both cases

Reply:
You're right, sorry for the noise.

Reply:
@biddisco: wouldn't be `sizeof(std::tuple<Data...>)` be sufficient here?